### ISSUE TITLE
Node reference id errors

### ISSUE THREAD

**Issue by pandurangpatil** [joernio/joern#3652]
```
Unhandled node_type - ast.Field -- 3482
Unhandled node_type - ast.TypeSpec -- 41
Unhandled node_type - ast.ValueSpec -- 5
Unhandled node_type - ast.FuncDecl -- 2
```

**Comment by pandurangpatil** [comment]
No need to handle `ast.Field` error, as we don't actually directly use this node but the child nodes of this node. Made fix to remove those warning with this pr - #3650

**Comment by ankit-privado** [comment]
ValueSpec issue example:
[WARN ] Unhandled node_type ast.ValueSpec node_id: 46 filename: Test0.go
```
package main

func main() {}

var person = Person()

type Name struct {
   name string
}

const (
 personName string = "peter"
)

func Person() Name {
   return Name{
     name: personName,
   }
}
```
FuncDecl Issue Example:
[WARN ] Unhandled node_type ast.FuncDecl node_id: 41 filename: Test0.go
```
package main

func help() {
   Person(Hello, "hi")
}

func Hello(value string) {}

func Person(hello func(string), value string) {
   hello(value)
}
```
